### DISCUSSION TITLE
Generic Energy Modbus driver

### DISCUSSION THREAD

**Discussion by msetina** [arendst/Tasmota#21630] [General]
I found a reference of generic Energy Modbus driver in https://github.com/arendst/Tasmota/blob/development/tasmota/energy_modbus_configs/configurations.md, but I am confused if this means that the tasmota build needs the same procedure as for Smart meter interface or does that utilize the same underpinings as Modbus  Bridge and is usable with stock builds?

**Comment by sfromis** [comment]
While I've not worked with that setup, what I *can* say is that it is unrelated to, and independent of the scripting-based Smart Meter Interface feature. AFAICT, you'll need a custom build with `#define USE_MODBUS_ENERGY`, along with `#define USE_ENERGY_SENSOR`, the latter already in most builds.
